File: src/modulo-01-java-OO/Modulo01/ElfoNoturno.java
public class ElfoNoturno extends Elfo {
    
    public ElfoNoturno(String nome) {
        super(nome);
    }
    
    @Override
    public void atirarFlecha(Dwarf dwarf) {
        super.atirarFlecha(dwarf);
        this.experiencia += 2;
        double percentual = this. vida * 0.05;
        this.vida -= percentual;
        this.status = (int)this.vida == 0 ? Status.MORTO : this.status;
    }
}

File: src/modulo-03-web/js/dia-04/ElfoNoturno.js
/* ======================================= *
 *	Sub-classe da classe Elfo - Herança    *
 * ======================================= */

function ElfoNoturno () {
	// super(nome, flechas);
	// Elfo.call(this, nome ,flechas);
	Elfo.apply(this, arguments);
};

// public class ElfoNoturno extends Elfo
ElfoNoturno.prototype = Object.create(Elfo.prototype);

// override
ElfoNoturno.protype.atirarFlecha = function() {
	// super.atirarFlecha();
	// Elfo.protoype.atirarFlecha.call(this, dwarf);
	Elfo.protoype.atirarFlecha.apply(this, arguments);
	console.log('sobrecarga do atirarFlecha');
};

// static
ElfoNoturno.mediaDeAltura = function() {
	return 2.15;
};